torch/utils/collect_env.py

```diff
@@ -113,15 +114,25 @@ def get_cudnn_version(run_lambda):
     if get_platform() == 'win32':
         cudnn_cmd = 'where /R "%CUDA_PATH%\\bin" cudnn*.dll'
     else:
-        cudnn_cmd = 'find /usr/local /usr/lib -type f -name "libcudnn*" 2> /dev/null'
```

Comment:
You should look for it in `$CUDA_HOME/lib`, `$CUDNN_LIBRARY` (direct path to so/dylib/dll file) an, `$CUDA_ROOT/lib` and `$CUDNN_ROOT/lib`.

Reply:
Instead, one should specify environment variables (PATH, DYLD_LIBRARY_PATH) correctly according to 
https://docs.nvidia.com/cuda/cuda-installation-guide-mac-os-x/index.html
and apply installation instructions of cudnn (in short, copy cudnn libraries to CUDA library path).
Then using CUDA_HOME, CUDNN_LIBRARY, etc would not be needed.

Reply:
PR #16968 fixes the `get_cudnn_version` for MacOSX where `ldconfig` is not available.